internal/service/alarms/internal/dictionary/dictionary.go

```diff
@@ -0,0 +1,187 @@
+		slog.Info("loaded rules for resource type", "ResourceType ID", res.resourceTypeID, "rules count", len(res.rules))
+	}
+
+	// TODO: Load future alarm dictionary struct
+	// TODO: Load alarm dictionary into DB
+}
+
+// getRulesForCluster fetches Prometheus rules for a cluster resource type
+func (r *AlarmDictionary) getRulesForCluster(ctx context.Context, version string) ([]*monitoringv1.PrometheusRule, error) {
+	cluster, err := r.getManagedCluster(ctx, version)
+	if err != nil {
+		return nil, err
+	}
+
+	rules, err := r.getRules(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return rules.Items, nil
+}
+
+// getManagedCluster finds a single managed cluster with the given version
+func (r *AlarmDictionary) getManagedCluster(ctx context.Context, version string) (*clusterv1.ManagedCluster, error) {
+	// Match managed cluster with the given version and not local cluster
+	selector := labels.NewSelector()
+	versionSelector, _ := labels.NewRequirement(managedClusterVersionLabel, selection.Equals, []string{version})
+	localClusterRequirement, _ := labels.NewRequirement(localClusterLabel, selection.NotEquals, []string{"true"})
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, clients.ListRequestTimeout)
+	defer cancel()
+
+	var managedClusters clusterv1.ManagedClusterList
+	err := r.Client.List(ctxWithTimeout, &managedClusters, &client.ListOptions{
+		LabelSelector: selector.Add(*versionSelector).Add(*localClusterRequirement),
+		Limit:         1,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error listing managed clusters: %w", err)
+	}
+
+	if len(managedClusters.Items) == 0 {
+		return nil, fmt.Errorf("no managed cluster found with version %s", version)
+	}
+
+	return &managedClusters.Items[0], nil
+}
+
+var getClientForCluster = clients.NewClientForCluster
+
+// getRules gets Prometheus rules from a managed cluster
+func (r *AlarmDictionary) getRules(ctx context.Context, cluster *clusterv1.ManagedCluster) (*monitoringv1.PrometheusRuleList, error) {
+	clusterClient, err := getClientForCluster(ctx, r.Client, cluster.Name)
+	if err != nil {
+		return nil, fmt.Errorf("error creating client for managed cluster %s: %w", cluster.Name, err)
+	}
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, clients.ListRequestTimeout)
+	defer cancel()
+
+	var rules monitoringv1.PrometheusRuleList
+	err = clusterClient.List(ctxWithTimeout, &rules)
+	if err != nil {
+		return nil, fmt.Errorf("error listing rules for managed cluster %s: %w", cluster.Name, err)
+	}
+
+	if len(rules.Items) == 0 {
+		return nil, fmt.Errorf("no rules found for managed cluster %s", cluster.Name)
+	}
+
+	return &rules, nil
+}
+
+// TODO: Delete once Resource Type is defined by Resource Server
+type resourceType struct {
+	id      int
+	model   string
+	version string
+}
+
+// TODO: Replace with actual resource type list from Resource Server
+func getResourceTypes() []resourceType {
+	return []resourceType{
+		{
+			id:      1,
```

Comment:
for all IDs let's use UUID

Reply:
This is throw away code. Everything around resource type will be modified at some point.

Reply:
just heads up when we upload this data into DB...it will reject anything that's not UUID. But we can update this in the follow up PR during DB integration.